src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -377,9 +392,98 @@ int fmi3_xml_set_attr_enum(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_
     return 0;
 }
 
-int fmi3_xml_set_attr_boolean(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_t elmID, fmi3_xml_attr_enu_t attrID, int required, unsigned int* field, unsigned int defaultVal) {
+int fmi3_xml_set_attr_boolean(fmi3_xml_parser_context_t *context, fmi3_xml_elm_enu_t elmID, fmi3_xml_attr_enu_t attrID,
+        int required, unsigned int* field, unsigned int defaultVal)
+{
     jm_name_ID_map_t fmi_boolean_i_dMap[] = {{"true", 1},{"false", 0}, {"1", 1},{"0", 0}, {0,0}};
-    return fmi3_xml_set_attr_enum(context,elmID, attrID,required, field, defaultVal, fmi_boolean_i_dMap);
+    return fmi3_xml_set_attr_enum(context, elmID, attrID, required, field, defaultVal, fmi_boolean_i_dMap);
+}
+
+// TODO: For FMI3, do we want to use bool in the getters for boolean attributes, or keep using unsigned int?
```

Comment:
That would mean that we would need to change the return value of some API, but also make code more consistent.

I mean, it would be good if we could change all boolean functions to return `bool`. Right now we mainly return `int` but at least the one for getting FMU capabilities return `unsigned int`, probably because that's how they were parsed with this fcn.

We prob need a meeting to discuss it but early input is welcome

Reply:
Okay good. I'll bring it up later for more discussion